plugins/field-colour-hsv-sliders/src/field_colour_hsv_sliders.ts

```diff
@@ -0,0 +1,639 @@
+    if (rgb.r == max) {
+      hue = (rgb.g - rgb.b) / delta;
+    } else if (rgb.g == max) {
+      hue = 2 + ((rgb.b - rgb.r) / delta);
+    } else {
+      hue = 4 + ((rgb.r - rgb.g) / delta);
+    }
+    hue /= 6;
+    this.h = hue - Math.floor(hue);
+    return this;
+  }
+
+  /**
+   * Updates the properties of this instance to copy the provided HsvColour.
+   * @param other An HSV representation of a colour to copy.
+   * @return This instance after updating it.
+   */
+  copy(other: HsvColour): HsvColour {
+    this.h = other.h;
+    this.s = other.s;
+    this.v = other.v;
+    return this;
+  }
+}
+
+/**
+ * Class for a colour input field that displays HSV slider widgets when clicked.
+ */
+export class FieldColourHsvSliders extends Blockly.FieldColour {
+  /* eslint-disable @typescript-eslint/naming-convention */
+  /** The maximum value of the hue slider range. */
+  private static readonly HUE_SLIDER_MAX: number = 360;
+
+  /** The maximum value of the saturation slider range. */
+  private static readonly SATURATION_SLIDER_MAX: number = 100;
+
+  /** The maximum value of the brightness slider range. */
+  private static readonly BRIGHTNESS_SLIDER_MAX: number = 100;
+
+  /**
+   * The gradient control point positions should align with the center of the
+   * slider thumb when the corresponding colour is selected. When the slider
+   * is at the minimum or maximum value, the distance of center of the thumb
+   * from the edge of the track will be the thumb's radius, so that's how far
+   * the minimum and maximum control points should be.
+   */
+  static readonly THUMB_RADIUS: number = 12;
+  /* eslint-enable @typescript-eslint/naming-convention */
+
+  /** Helper colour structures to allow manipulation in the HSV colour space. */
+  private static readonly helperHsv: HsvColour = new HsvColour();
+
+  /** Helper colour structures to support conversion to the RGB colour space. */
+  private static readonly helperRgb: RgbColour = new RgbColour();
+
+  /** Array holding info needed to unbind events. Used for disposing. */
+  private boundEvents: Blockly.browserEvents.Data[] = [];
+
+  /** HTML span element to display the current hue. */
+  private hueReadout: HTMLSpanElement | null = null;
+
+  /** HTML range input element for editing hue. */
+  private hueSlider: HTMLInputElement | null = null;
+
+  /** HTML span element to display the current saturation. */
+  private saturationReadout: HTMLSpanElement | null = null;
+
+  /** HTML range input element for editing saturation. */
+  private saturationSlider: HTMLInputElement | null = null;
+
+  /** HTML span element to display the current brightness. */
+  private brightnessReadout: HTMLSpanElement | null = null;
+
+  /** HTML range input element for editing brightness. */
+  private brightnessSlider: HTMLInputElement | null = null;
+
+  /** HTML div element containing all the labels and sliders. */
+  private dropdownContainer: HTMLDivElement | null = null;
+
+  /**
```

Comment:
Do we need this constructor at all if it's just a passthrough for the superclass?

Reply:
Removed.